hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java

```diff
@@ -1166,4 +1172,107 @@ public String getClusterId() {
   public HDDSLayoutVersionManager getLayoutVersionManager() {
     return scmLayoutVersionManager;
   }
+
+  private void waitForAllContainersToClose() {
```

Comment:
This may become a costly operation during finalize. If we have destroyed all pipelines, the containers are automatically closed. So, we maybe able to skip this.

Reply:
Yes, but shouldnt we wait for all containers to close before we finalize the upgrade ? Because as part of closing the containers wemay excercise older layout version of the code otherwirse.

Reply:
Agree the point that @avijayanhwx  mentioned. With the number of containers/pipeline becomes more large, this operation cal be very costly. can we just add a flag switch to avoid write type ops for container/pipelines, like freezePipelineCreation we added above.

Reply:
Agreed and removed the "waitForAllContainersToClose()" call. Leaving the function around for now just in case we need it in future.